staging/src/k8s.io/component-base/cli/flag/sectioned_test.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMergeNamedFlagSets(t *testing.T) {
```

Comment:
I think this could use a couple additional unit tests cases:
* What if one or both of the NamedFlagSets is empty?
* What if there is a name collision? Both NamedFlagSets have a different FlagSet with the same name.
* Can you add some flags in the flag sets to make them a little more realistic?  That way you can assert not only that a FlagSet exists with the right name, but that it has the expected flags.

Reply:
- Added test case for one or both of the NamedFlagSets to be empty.
- I've added a name collision test case, however for now is assuming silently overwriting is desired, I suspect that is not what we want, if we return a error in `controllermanager` it would not be ultimately handle, we could panic... WDYT?
- I've added a few flags in the flag sets then asserted on their existence.